### DISCUSSION TITLE
Getting avg of relation field

### DISCUSSION THREAD

**Discussion by lgraubner** [prisma/prisma#10796] [Q&A] [answered]
Hi!

I'm trying to get all related entities from an parent entity including the average of one of it's fields. I can't find any way on how to do this. For clarification some code:

My schema looks like this:

```prisma
model Course {
  id          Int       @id @default(autoincrement())
  name        String    @db.VarChar(255)
  slug        String    @db.VarChar(255) @unique
  reviews     Review[]
  topics      Topic[]
  createdAt   DateTime  @default(now())
  updatedAt   DateTime  @updatedAt
}

model Review {
  id          Int       @id @default(autoincrement())
  rating      Int
  course      Course    @relation(fields: [courseId], references: [id])
  courseId    Int
  createdAt   DateTime  @default(now())
  updatedAt   DateTime  @updatedAt
}
```

And I'm trying to query like this:

```ts
let courses = await db.course.findMany({
  where: {
    topics: {
      some: {
        slug: {
          equals: topic.slug,
        },
      },
    },
  },
  include: {
    reviews: true,
});
```

This works fine. Now I'd like to get the average rating of each courses review to be included. But as far as I can see I can only get the average when directly querying the Review entity. Is there any way to do this?

Thanks in advance!

**Comment by lgraubner** [comment]
What works as second query is the following:

```ts
let avg = await db.review.groupBy({
  by: ['courseId'],
  where: {
    course: {
      topics: {
        some: {
          slug: {
            equals: topic.slug,
          },
        },
      },
    },
  },
  _avg: {
    rating: true,
  },
});
```

I can map those results with the first query. But is it possible to achieve the same as a single query?

**Comment by austincrim** [comment, marked as answer]
Hey @lgraubner 👋 ,

I don't think it is possible to do both operations in a single query without using SQL and [`$queryRaw`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#queryraw).

Would you mind opening a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with these details for our engineering teams to consider?

Thanks!

**Reply by lgraubner** [reply]
Done! (#10866) Thanks for the answer though.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
